core/domain/activity_jobs_one_off_test.py

```diff
@@ -2137,3 +2137,291 @@ def test_missing_topic_rights_commit_logs(self):
+        stringified_output = (
+            activity_jobs_one_off.AddMissingCommitLogsJob
+            .get_output(job_id))
+
+        eval_output = [
+            ast.literal_eval(stringified_item) for
+            stringified_item in stringified_output]
+        return eval_output
+
+    def test_validate_model_names_list(self):
+        job_class = activity_jobs_one_off.AddMissingCommitLogsJob
+        class_names = {
+            cls.__name__ for cls in (
+                job_class.SNAPSHOT_METADATA_MODELS_WITH_MISSING_COMMIT_LOGS)}
+        model_names_with_default_commit_status = set(
+            job_class.MODEL_NAMES_WITH_DEFAULT_COMMIT_STATUS)
+        model_names_with_commit_status_in_rights = set(
+            job_class.MODEL_NAMES_WITH_COMMIT_STATUS_IN_RIGHTS)
+        aggregate_model_names = (
+            model_names_with_default_commit_status |
+            model_names_with_commit_status_in_rights)
+        common_model_names = (
+            model_names_with_default_commit_status &
+            model_names_with_commit_status_in_rights)
+
+        self.assertEqual(len(common_model_names), 0)
+        self.assertItemsEqual(class_names, aggregate_model_names)
+        self.assertItemsEqual(
+            class_names, set(job_class.MODEL_NAMES_TO_PROPERTIES.keys()))
+
+    def test_delete_exp_rights_related_models(self):
+        exp_rights = exp_models.ExplorationRightsModel(
+            id=self.EXP_ID,
+            status='public',
+            version=1,
+            deleted=True
+        )
+        base_models.BaseModel.put_multi([exp_rights])
+        exp_models.ExplorationRightsSnapshotMetadataModel(
+            id='%s-1' % self.EXP_ID,
+            committer_id=self.albert_id,
+            commit_type='edit',
+            commit_cmds=self.DUMMY_COMMIT_CMDS,
+        ).put()
+
+        expected_output = [
+            [
+                'SUCCESS-Parent model marked deleted-Deleted all ' +
+                'related models-ExplorationRightsSnapshotMetadataModel',
+                ['exp_id0-1']
+            ]
+        ]
+
+        actual_output = self._run_one_off_job()
+
+        parent_model = exp_models.ExplorationRightsModel.get_by_id(
+            self.EXP_ID)
+        snapshot_model = (
+            exp_models.ExplorationRightsSnapshotMetadataModel.get_by_id(
+                '%s-%s' % (self.EXP_ID, 1)))
+
+        self.assertIsNone(parent_model)
+        self.assertIsNone(snapshot_model)
+        self.assertItemsEqual(expected_output, actual_output)
+
+    def test_add_missing_exp_rights_commit_logs(self):
+        exp_rights = exp_models.ExplorationRightsModel(
+            id=self.EXP_ID,
+            status='public'
+        )
+        base_models.BaseModel.put_multi([exp_rights])
+        content_dict = {
+            'status': 'public',
+            'owner_ids': self.albert_id,
+            'editor_ids': self.albert_id,
+            'voice_artist_ids': self.albert_id,
+            'viewer_ids': self.albert_id
+
```

Comment:
Nit: remove unnecessary newline for consistency.

Reply:
Done.